app/Http/Search/Request.php

```diff
@@ -638,6 +645,33 @@ public function addFunctionScore($params, $input)
         return $params;
     }
 
+    /**
+     * Add param for vector search
+     *
+     * @link https://www.elastic.co/guide/en/elasticsearch/reference/6.0/query-dsl-function-score-query.html
+     *
+     * @param array $params
+     *
+     * @return array
+     */
+    public function addScriptParam($params, $input)
+    {
+        if ($input['q']) {
+            $queryVector = app('Embeddings')->getEmbeddings($input['q']);
+
+            $params['body']['query']['script_score']['script'] = [
+                'source' => "if (doc['text_embedding'].size() == 0) { return 0; } else { return cosineSimilarity(params.query_vector, 'text_embedding') + 1.0; }",
```

Comment:
Could you leave a comment that describes what this code is doing: `cosineSimilarity(params.query_vector, 'text_embedding') + 1.0;`